libs/auth/src/inserted_smart_card_auth.ts

```diff
@@ -184,7 +179,7 @@ export class InsertedSmartCardAuth implements InsertedSmartCardAuthApi {
 
   async clearCardData(): Promise<Result<void, Error>> {
     try {
-      await this.card.writeLongUint8Array(Uint8Array.of());
+      await this.card.writeData(Buffer.from([]));
```

Comment:
Should the Card API have a `clearData` method? I'm thinking that since it has a `clearUserAndData` method, the concept of clearing data is already defined behind the Card abstraction, so it's possible that this way of clearing data that the consumer is doing here (writing an empty buffer) might not align with the way that `clearUserAndData` is doing it.

In other words, who gets to decide how clearing data is implemented - the Card API or its consumer?

Related: there's a bit of a conflation of an empty buffer and "no data". Are we good with that? I don't think we have any reason we'd want to support explicitly writing an empty buffer as valid data, but just thought I'd check.

Reply:
Ooh good call about ensuring consistency between standalone clearing of data and clearing of both the user and data! Will add a `clearData` method to the card API.

To the second point, as far as I can tell, under the hood, an empty buffer and "no data" are equivalent. `Buffer.from([])` basically means no bytes of data.